src/device/wg_controller.rs

```diff
@@ -54,6 +54,12 @@ pub async fn consolidate_wg_state(
     entities: &Entities,
     features: &Features,
 ) -> Result {
+    if features.post_quantum_vpn.is_some() && requested_state.postquantum_keys.is_none() {
+        // The post quantum state is not ready, we don't want to set up quantum
+        // unsafe tunnel
+        return Ok(());
+    }
+
```

Comment:
Please clarify my understanding on this one - `requested_state` is the request we send to WG driver and it will try it's best to match it?

In this case if there's a request for VPN connection but PQ keys are not present it will result in this call being prematurely exited and thus WG will not be set up for the spcified VPN server? TLDR: having no PQ keys doesn't allow for VPN connection?

Reply:
The requested_state structure is specific to telio device and holds the full information of current configuration. It's a big mutable structure and when something changes then a certain part of it is modified so that other parts of the system are aware of the change by consulting the structure.

So yes, when the PQ keys are not set up then the wg interface is not configured. However that's expected, because configuring the interface means setting up a quantum unsafe tunnel and we don't want that. Keep in mind that this call is actively executed when the keys are available.

Reply:
Does it still allow for non-pq vpn connections?

Reply:
No. And it shouldn't.

Reply:
Oh my bad. After looking again and again I saw that it's
```
if features.post_quantum_vpn.is_some
```
and not
```
if features.vpn.is_some
```

:D